pygrocy/grocy_api_client.py

```diff
@@ -327,39 +327,59 @@ def __init__(self, base_url, api_key, port: int = DEFAULT_PORT_NUMBER, verify_ss
                 "GROCY-API-KEY": api_key
             }
 
+    def _do_request(self, request_type: str, end_url: str, data = None):
```

Comment:
I think I would favor a separate method for each HTTP method, e.g. `_do_get_request`, `_do_post_request`...